cmd/ferretdb/main.go

```diff
@@ -19,37 +19,62 @@ import (
 	"flag"
 	"fmt"
 	"os"
-	"time"
+	"strings"
 
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/common/expfmt"
 	"go.uber.org/zap"
+	"golang.org/x/exp/maps"
+	"golang.org/x/exp/slices"
 
 	"github.com/FerretDB/FerretDB/internal/clientconn"
 	"github.com/FerretDB/FerretDB/internal/handlers"
-	"github.com/FerretDB/FerretDB/internal/handlers/dummy"
-	"github.com/FerretDB/FerretDB/internal/handlers/pg"
-	"github.com/FerretDB/FerretDB/internal/handlers/pg/pgdb"
 	"github.com/FerretDB/FerretDB/internal/util/debug"
 	"github.com/FerretDB/FerretDB/internal/util/logging"
+	"github.com/FerretDB/FerretDB/internal/util/must"
 	"github.com/FerretDB/FerretDB/internal/util/version"
 )
 
-//nolint:gochecknoglobals // flags are defined there to be visible in `bin/ferretdb-testcover -h` output
+// newHandlerOpts represents configuration for all handlers.
+//
+// Some fields may be unused by some handlers.
+type newHandlerOpts struct {
+	ctx    context.Context
+	logger *zap.Logger
+}
+
+// newHandler is a function that constructs a new handler.
+type newHandler func(opts *newHandlerOpts) (handlers.Interface, error)
+
+// registeredHandlers map handler names for constructors.
+var registeredHandlers = map[string]newHandler{}
+
 var (
-	debugAddrF       = flag.String("debug-addr", "127.0.0.1:8088", "debug address")
-	listenAddrF      = flag.String("listen-addr", "127.0.0.1:27017", "listen address")
-	modeF            = flag.String("mode", string(clientconn.AllModes[0]), fmt.Sprintf("operation mode: %v", clientconn.AllModes))
-	postgresqlURLF   = flag.String("postgresql-url", "postgres://postgres@127.0.0.1:5432/ferretdb", "PostgreSQL URL")
-	proxyAddrF       = flag.String("proxy-addr", "127.0.0.1:37017", "")
-	versionF         = flag.Bool("version", false, "print version to stdout (full version, commit, branch, dirty flag) and exit")
+	versionF = flag.Bool("version", false, "print version to stdout (full version, commit, branch, dirty flag) and exit")
+
+	listenAddrF = flag.String("listen-addr", "127.0.0.1:27017", "listen address")
+	proxyAddrF  = flag.String("proxy-addr", "127.0.0.1:37017", "proxy address")
+	debugAddrF  = flag.String("debug-addr", "127.0.0.1:8088", "debug address")
+	modeF       = flag.String("mode", string(clientconn.AllModes[0]), fmt.Sprintf("operation mode: %v", clientconn.AllModes))
+
+	handlerF       = flag.String("handler", "pg", "<set in initFlags()>")
```

Comment:
Can we add a comment that it's set in *`init` functions for the handler? 
And rename handler initialization files to `handler_dummy.go`, `handler_pg.go`,  `handler_tigris.go`?